### PR TITLE
Handling mixtures more elegantly, adding useful enums

### PR DESCRIPTION
**Pull request by connorcoley**
Following this morning's discussion...
- "addition_stage" -> "addition_order" // 1-indexed
- How to differentiate all flavors of NMR? [in details]
- How to allow stock solutions with more than one solute? [now using ```components``` field of Compound message; please review]
- Compound preparation enum: add COMPOUND_PREPARATION_SYNTHESIZED = 5;
- Add glass vial to ReactionSetup enum, since it is very common

### CODE REVIEW

**Comment by connorcoley** [comment]
Latest commit addresses #15

File: proto/ord_schema.proto [hunk 1]
```diff
@@ -40,73 +40,174 @@ message Reaction {
...
   AdditionSpeed addition_speed = 6;
 }
 
 message Compound {
+  // Set of identifiers used to uniquely define this compound.
+  // Solutions or mixed compounds should use the NAME identifier
+  // and list all constituent compounds in the "components" field.
   repeated CompoundIdentifier identifiers = 1;
+  // Quantity of this compound used in the reaction.
   Amount amount = 2;
```
Comment (skearnes) [inline proto/ord_schema.proto]: Should amount/components be a Oneof?

File: proto/ord_schema.proto [hunk 2]
```diff
@@ -186,74 +284,21 @@ message Amount {
...
 message ReactionSetup {
   enum VesselType {
     VESSEL_TYPE_UNKNOWN = 0;
     VESSEL_TYPE_ROUND_BOTTOM_FLASK = 1;
-    VESSEL_TYPE_WELL_PLATE = 2;
-    VESSEL_TYPE_MICROWAVE_VIAL = 3;
-    VESSEL_TYPE_TUBE = 4;
-    VESSEL_TYPE_CONTINUOUS_STIRRED_TANK_REACTOR = 5;
-    VESSEL_TYPE_PACKED_BED_REACTOR = 6;
+    VESSEL_TYPE_VIAL = 2;  // Regular glass vial
```
Comment (skearnes) [inline proto/ord_schema.proto]: Useful to distinguish between glass/plastic?

File: proto/ord_schema.proto [hunk 3]
```diff
@@ -599,114 +660,126 @@ message Person {
  */
 message Time {
   enum TimeUnit {
-    TIME_UNIT_HOUR = 0;
-    TIME_UNIT_MINUTE = 1;
-    TIME_UNIT_SECOND = 2;
+    TIME_UNIT_UNKNOWN = 0;
```
Comment (skearnes) [inline proto/ord_schema.proto]: Why the UNKNOWN values for these units? Is this just to have a better default? If so, that's fine but I wanted to check.

File: proto/ord_schema.proto (hunk 1 above)
Comment (connorcoley) [reply proto/ord_schema.proto]: I envision a use-case where "amount" is used to specify the volume of a stock solution that is added, while "components" specifies the make-up of the solution, whose total volume can exceed the amount that was added in this particular reaction.

File: proto/ord_schema.proto (hunk 2 above)
Comment (connorcoley) [reply proto/ord_schema.proto]: Yes, good point. Would a separate enum be overkill? Not all plastics are equal, but I do think that ```VESSEL_MATERIAL_xxxx``` could still be useful

File: proto/ord_schema.proto (hunk 3 above)
Comment (connorcoley) [reply proto/ord_schema.proto]: Yes, sorry for not documenting the reason/issue well

**Pull request merged**